File: python/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

#%%
from typing import List

#%%
# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = dict()
        for idx, num in enumerate(nums):
            if num in hash:
                return [hash[num], idx]
            else:
                diff = target - num
                hash[diff] = idx
# @lc code=end

#%%
print(Solution().twoSum([2,7,15,19], 9))

File: csharp/1.two-sum.cs
/*
 * @lc app=leetcode id=1 lang=csharp
 *
 * [1] Two Sum
 */
 
Console.WriteLine($"[{String.Join(", ", new Solution().TwoSum(new int[] {2,7,15,19}, 9))}]");

// @lc code=start
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        for(int i = 0; i < nums.Length; i++) {
            int diff = target - nums[i];
            for(int j = i+1; j < nums.Length; j++) {
                if(nums[j] == diff)
                    return new[] { i, j };
            }
        }
        return null;
    }
}
// @lc code=end

